app/models/folio/group.rb

```diff
@@ -8,16 +8,17 @@ def patron_type
     end
 
     def sponsor
-      nil
-    end
+      return self if sponsor?
```

Comment:
Shouldn't this be cached in @sponsor too? Then you don't have to repeat this check many times?